owa/app/js/subscription/subscription.controller.js

```diff
@@ -48,6 +52,27 @@ class SubscriptionController {
         vm.subscription = response.results[0];
       })
     }
+
+    /*
+    *This checks if collection version has been passed to subscription url by checking number of forward slashes after https://api.openconceptlab.org
+    *If the number is 5, such as with https://api.openconceptlab.org/users/username/collections/collectionname/v1.0
+    *that means collection version was passed and isVersionAdded() will return true
+    */
+    function isVersionAdded(subscriptionUrl) {
+      if (subscriptionUrl.endsWith("/")) {
+        subscriptionUrl = subscriptionUrl.substring(0, subscriptionUrl.lastIndexOf('/'));
+      }
+      let words = subscriptionUrl.split(OCL_API_BASE_URL)
```

Comment:
On the above line, are you trying to get a list of tokens where OCL_API_BASE_URL is the separator? Or are you simply trying to get the string after OCL_API_BASE_URL? If the latter, then i find the use of `split` confusing.

Reply:
Had tried using subString function before but let to more confusing code like 

w = word.substring(word.lastIndexOf('org') + 3) 

Do you have a better solution for this

Reply:
Ok tell me what you are trying to do with subscriptionUrl.split(OCL_API_BASE_URL) and then i will advise you accordingly.

Reply:
To get the sub string after OCL_API_BASE_URL

Reply:
Wouldn't this work?

subscriptionUrl.substring(OCL_API_BASE_URL.length);

Reply:
Actually that is return the whole string as it is

Reply:
waoh thank you